Content.Server/GameObjects/Components/Atmos/GasTankComponent.cs

```diff
@@ -1,10 +1,323 @@
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Map;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+using Robust.Shared.ViewVariables;
 
 namespace Content.Server.GameObjects.Components.Atmos
 {
     [RegisterComponent]
-    public class GasTankComponent : Component
+    [ComponentReference(typeof(IActivate))]
+    public class GasTankComponent : SharedGasTankComponent, IExamine, IGasMixtureHolder, IUse, IDropped, IActivate
     {
-        public override string Name => "GasTank";
+    	private const float MaxExplosionRange = 14f;
+        private const float DefaultOutputPressure = Atmospherics.OneAtmosphere;
+
+        private float _pressureResistance;
+
+        private int _integrity = 3;
+
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [ViewVariables] private BoundUserInterface? _userInterface;
+
+        [ViewVariables] public GasMixture? Air { get; set; }
+
+        /// <summary>
+        /// Maximum output pressure when valve is open. When tank is controlled
+        /// by internals component it has no effect.
+        /// </summary>
+        [ViewVariables] public float OutputPressure { get; private set; }
+
+        /// <summary>
+        /// Tank is connected to external system and valve is controlled automatically
+        /// </summary>
+        [ViewVariables] public bool IsConnected { get; set; }
+
+        /// <summary>
+        /// Represents that tank is functional and can be connected to internals
+        /// </summary>
+        public bool IsFunctional => GetInternalsComponent() != null;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _userInterface = Owner.GetUIOrNull(SharedGasTankUiKey.Key);
+            if (_userInterface != null)
+            {
+                _userInterface.OnReceiveMessage += UserInterfaceOnOnReceiveMessage;
+            }
+        }
+
+        public void OpenInterface(IPlayerSession session)
+        {
+            _userInterface?.Open(session);
+            UpdateUserInterface(true);
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataReadWriteFunction("air", new GasMixture(), x => Air = x, () => Air);
+            serializer.DataReadWriteFunction("outputPressure", DefaultOutputPressure, x => OutputPressure = x,
+                () => OutputPressure);
+            serializer.DataField(ref _pressureResistance, "pressureResistance", Atmospherics.OneAtmosphere * 5f);
+        }
+
+        public void Examine(FormattedMessage message, bool inDetailsRange)
+        {
+            message.AddMarkup(Loc.GetString("Pressure: [color=orange]{0}[/color] kPa.",
+                Math.Round(Air?.Pressure ?? 0)));
+            if (IsConnected)
+            {
+                message.AddMarkup(Loc.GetString("\nConnected to external component"));
```

Comment:
Should the `\n` be brought out of the loc?

Reply:
true, brought it to the one above